modules/KalturaSupport/resources/mw.kAnalony.js

```diff
@@ -540,8 +540,55 @@
 			}
 			return "";
 		},
+		
+		getForwardBufferHealth : function(){
+			var forwardBufferHealth = NaN;
+			try{
+				var availableBuffer = this.availableBuffer();
+				var targetBuffer = this.getTargetBuffer();
+				if (targetBuffer && availableBuffer) {
+					// considering playback left to the target calculation
+					forwardBufferHealth = Math.round((availableBuffer * 1000) / targetBuffer) / 1000;
+				}
+				mw.log("forwardBufferHealth  "+forwardBufferHealth);
+			}catch(e){
+				mw.log("Failed getting getForwardBufferHealth data");
+			}
+			return forwardBufferHealth;
+		},
+		
 
+		
+		// samples the hls at realtime 
+		getTargetBuffer : function(){
+			var targetBuffer = null;
+			if(this.embedPlayer.plugins.hlsjs){
+				targetBuffer = this.embedPlayer.plugins.hlsjs.targetBuffer();
```

Comment:
to avoid this coupling create the interface methods on the embedPlayer and return correct default values there and get the plugin that implements the methods to return correct values according to its implementation